resolver/resolver.cc

```diff
@@ -605,7 +615,19 @@ class ResolveConstantsWalk {
         for (auto &ancst : original.ancestors) {
             bool isSuperclass = (original.kind == ast::ClassDef::Kind::Class && &ancst == &original.ancestors.front() &&
                                  !klass.data(ctx)->isSingletonClass(ctx));
-            transformAncestor(isSuperclass ? ctx : ctx.withOwner(klass), klass, ancst, isSuperclass);
+            transformAncestor(isSuperclass ? ctx : ctx.withOwner(klass), klass, ancst,
+                              isSuperclass ? AncestorResolutionItem::Kind::Superclass
+                                           : AncestorResolutionItem::Kind::Mixin);
+        }
+
+        // Each time the classdef is reloaded we need to clear the list of requiredAncestors
+        // before loading them again.
+        auto sym = klass.data(ctx);
+        sym->requiredAncestorsLocs.clear();
```

Comment:
This doesn't look right. Deleting things from the symbol table is almost never what you want.

I think this breaks with a setup like this:

```ruby
# -- a_b.rb --
class A; end
class B; end

# -- m.rb --
module M
  requires_ancestor A
end

# -- n.rb --
module N
  requires_ancestor B
end
```

and then providing an edit to `m.rb` to add a blank line at the end. That would trigger the fast path, but all the `requiredAncestorLocs` would have been cleared, and only `A` will be added back as a required ancestor.